### PR TITLE
docs: adds section about DB session management

### LINKED ISSUE #123: docs: add reference about session management
1. Add [this documentation](https://invenio-talk.web.cern.ch/t/understanding-database-session-management-and-transactions-in-invenio/33) to the module in a similar way to how the [optimistic concurrency control](https://invenio-records.readthedocs.io/en/latest/concurrency.html) was done in Invenio-Records.

2. Add a cross-reference to the "optimistic concurrency control" in Invenio-Records

### PR DESCRIPTION
**Pull request by roksys**
* Closes #123

### CODE REVIEW

File: docs/sessions_management.rst [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+Database session management
+===========================
+
+Invenio uses `SQLAlchemy Toolkit and Object Relational Mapper <https://www.sqlalchemy.org>`_
+for all database related operations.
+
+
+Transactions are tied to requests
+---------------------------------
+In Invenio, a transaction is tied to a HTTP request. By default, the transaction is
```
Comment (equadon) [inline docs/sessions_management.rst]:
```suggestion
In Invenio, a transaction is tied to an HTTP request. By default, the transaction is
```

File: docs/sessions_management.rst [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+Transactions are tied to requests, because the outer view, in charge of handling the
+request, needs full control of when a transaction is committed. If the view was not
+in charge, you could end up with inconsistent data in the database - for instance
+persistent identifier may have been committed, but the associated record was not committed.
+That is why Invenio makes use of SQLAlchemy’s version counter feature to provide `optimistic
+concurrency control (OCC) <https://invenio-records.readthedocs.io/en/latest/concurrency.html>`_
+on the records table when the database transaction isolation level is below repeatable read
+isolation level.
+
+When are SQL statements sent to the database server.
```
Comment (equadon) [inline docs/sessions_management.rst]:
```suggestion
When are SQL statements sent to the database server?
```

File: docs/sessions_management.rst [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+useful to force integrity constraints to be checked by the database,
+which may be needed by the database.
+
+What happens with exceptions in nested transactions?
+----------------------------------------------------
+If an exception occurs in a nested transaction, first the save point will be
+rolled back, and afterwards the entire transaction will be rolled back unless
+the exception is caught.
+
+For instance in the following code example, the entire transaction will be rolled backed:
```
Comment (equadon) [inline docs/sessions_management.rst]:
```suggestion
For instance in the following code example, the entire transaction will be rolled back:
```

File: docs/sessions_management.rst [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+    @app.route('/')
+    def index():
+        # db operations 1 ....
+        with db.session.begin_nested():
+            # db operations 2 ....
+            raise Exception()
+        db.session.commit()
+
+On the other, in the following example, the propagation of the exception is stopped,
```
Comment (equadon) [inline docs/sessions_management.rst]: On the other.. what? On the other hand?

File: docs/sessions_management.rst [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    @app.route('/')
+    def index():
+        # db operations 1 ....
+        with db.session.begin_nested():
+            # db operations 2 ....
+            raise Exception()
+        db.session.commit()
+
+On the other hand, in the following example, the propagation of the exception is stopped,
+and only the db 2 operations are rolled back, while db operations 1 are committed to
```
Comment (mvidalgarcia) [inline docs/sessions_management.rst]:
```suggestion
and only the db operations 2 are rolled back, while db operations 1 are committed to
```

File: docs/sessions_management.rst [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+the database.
+
+
+.. code:: python
+
+   @app.route('/')
+   def index():
+       # db operations 1 ....
+       try:
+           db.session.begin_nested():
```
Comment (mvidalgarcia) [inline docs/sessions_management.rst]:
```suggestion
           with db.session.begin_nested():
```

**Pull request merged**